### PR TITLE
chore: add generate:types command to lint-staged

### PR DESCRIPTION
**Pull request by bluwy**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
     "release": "pnpm changeset publish",
     "prepare": "husky",
     "playwright": "playwright-core",
-    "generate:types": "pnpm --filter \"./packages/*\" --parallel generate:types"
+    "generate:types": "pnpm --filter \"./packages/*\" --parallel generate:types",
+    "generate:types-staged": "pnpm generate:types && :"
```
Comment (bluwy) [inline package.json:23]: This separate command was needed as lint-staged will passed changed files to the command by default, and the `:` will toss them away. Don't know if Windows will be happy about this though.

I can't inline this directly into lint-staged, it seems to not like the `&&`.

Comment (dominikg) [reply package.json:23]: hmmm, can you consume the input in the generate script instead?

Comment (bluwy) [reply package.json:23]: Do you mean putting the `&& :` directly in `generate:types` instead? Yeah that works too if you're ok with that.

Comment (dominikg) [reply package.json:23]: no, i meant putting it in a node script that handles it, calling dts-buddy from there.

**Review by dominikg — APPROVED** [review]

**Pull request merged**